### DISCUSSION TITLE
Schedule re-archive - versioning

### DISCUSSION THREAD

**Discussion by dunnno** [ArchiveBox/ArchiveBox#687] [Q&A]
Hi there :)

I'm scratching my head on how to setup a schedule for ArchiveBox to scan again the websites i currently archive.

I'm using that command to each website i want a new archive version every monday at 2 AM : 
`archivebox schedule --every="0 2 * * 1" --add https://www.6th-domain.com/`

I've added a total of six websites, 5 of those have been added manually and scanned once, via the webUI. That previous line being the 6th one, not added in the webUI, thus not scanned yet.

Here is the result of the command :

> [√] Scheduled new ArchiveBox cron job for user: archivebox (6 jobs are active).
>     0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.1st-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
>     0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.2nd-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
>     0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.3rd-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
>     0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.4th-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
>     0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.5th-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
>    0 2 * * 1 cd /data && /usr/local/bin/archivebox add --depth=0 "https://www.6th-domain.com/" > /data/logs/archivebox.log 2>&1 # archivebox_schedule
> 
> [!] With the current cron config, ArchiveBox is estimated to run >312 times per year.
>     Congrats on being an enthusiastic internet archiver! 👌
> 
>     Make sure you have enough storage space available to hold all the data.
>     Using a compressed/deduped filesystem like ZFS is recommended if you plan on archiving a lot.

The first five schedules were added 2 days ago, and never fired up.

What is it that i'm doing wrong here ?

ArchiveBox 0.5.6

**Comment by pirate** [comment]
Without the `--overwrite` flag or `--depth=1` flag, re-archiving an existing domain that you've previously archived is a no-op.
archivebox does not support archiving a single site multiple times without changing the url https://github.com/ArchiveBox/ArchiveBox/issues/179.

Can you post the full output of `archivebox --version` and the contents of `/data/logs/archivebox.log`.

**Reply by dunnno** [reply]
Here you go :) : 

archivebox --version : 
 ```
ArchiveBox v0.5.6
Cpython Linux Linux-5.10.21-Unraid-x86_64-with-glibc2.28 x86_64 (not in Docker)

[i] Dependency versions:

√  ARCHIVEBOX_BINARY     v0.5.6          valid     /usr/local/bin/archivebox
 √  PYTHON_BINARY         v3.9.1          valid     /usr/local/bin/python3.9
 √  DJANGO_BINARY         v3.1.3          valid     /usr/local/lib/python3.9/site-packages/django/bin/django-admin.py
 √  CURL_BINARY           v7.64.0         valid     /usr/bin/curl
 √  WGET_BINARY           v1.20.1         valid     /usr/bin/wget
 √  NODE_BINARY           v15.8.0         valid     /usr/bin/node
 X  SINGLEFILE_BINARY     ?               invalid   single-file
 X  READABILITY_BINARY    ?               invalid   readability-extractor
 X  MERCURY_BINARY        ?               invalid   mercury-parser
 √  GIT_BINARY            v2.20.1         valid     /usr/bin/git
 √  YOUTUBEDL_BINARY      v2021.02.04.1   valid     /usr/local/bin/youtube-dl
 √  CHROME_BINARY         v88.0.4324.146  valid     /usr/bin/chromium
 √  RIPGREP_BINARY        v0.10.0         valid     /usr/bin/rg

[i] Source-code locations:
 √  PACKAGE_DIR           23 files        valid     /app/archivebox
 √  TEMPLATES_DIR         3 files         valid     /app/archivebox/templates

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled
 -  COOKIES_FILE          -               disabled

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data
 √  SOURCES_DIR           7 files         valid     ./sources
 √  LOGS_DIR              0 files         valid     ./logs
 √  ARCHIVE_DIR           8 files         valid     ./archive
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf
 √  SQL_INDEX             248.0 KB        valid     ./index.sqlite3

[!] Warning: Missing 3 recommended dependencies
    ! SINGLEFILE_BINARY: single-file (unable to detect version)
      Hint: npm install --prefix . "git+https://github.com/ArchiveBox/ArchiveBox.git"
            or archivebox config --set SAVE_SINGLEFILE=False to silence this warning

    ! READABILITY_BINARY: readability-extractor (unable to detect version)
      Hint: npm install --prefix . "git+https://github.com/ArchiveBox/ArchiveBox.git"
            or archivebox config --set SAVE_READABILITY=False to silence this warning

    ! MERCURY_BINARY: mercury-parser (unable to detect version)
      Hint: npm install --prefix . "git+https://github.com/ArchiveBox/ArchiveBox.git"
            or archivebox config --set SAVE_MERCURY=False to silence this warning
```

Can't find the log :/
```
# cat /data/logs/archivebox.log
cat: /data/logs/archivebox.log: No such file or directory
```

So, how can i delete the cron jobs i created in order to re make them ?

**Reply by pirate** [reply]
`archivebox schedule --clear `should do the trick to clear it.

It's strange that the logs file is missing, perhaps your cronjobs are not running at all? Do you have cron enabled on this system? Can you check your system log for cron failures? `grep -A 10 cron /var/log/syslog`

Also side note: I highly highly recommend installing Singlefile, it's by far the best archiving method out of all of them, and you're missing out on the highest fidelity archives without it.

**Reply by dunnno** [reply]
Cronjobs looks fine, the othe jobs run well afaik.

attempt to install Singlefile throws this : 

```
$ npm install --prefix . "git+https://github.com/ArchiveBox/ArchiveBox.git"
[..................] - idealTree:data: sill idealTree buildDeps
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise@4.2.6: request-promise has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm notice
npm notice New minor version of npm available! 7.5.1 -> 7.7.6
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.7.6
npm notice Run npm install -g npm@7.7.6 to update!
npm notice
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /data/node_modules/archivebox/node_modules/.bin/mercury-parser
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/data/node_modules/archivebox/node_modules/.bin/mercury-parser'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/archivebox/.npm/_logs/2021-03-31T15_27_37_762Z-debug.log
$ -sh: 3:  : not found
```

**Reply by pirate** [reply]
Can you check the sys log for cron errors related to archivebox `grep -A 10 archivebox /var/log/syslog`.

Also for singlefile, `rm -Rf package.json package-lock.json node_modules` in your data folder and then try this instead: 
`npm install --no-save --no-audit --no-fund --prefix . 'git+https://github.com/gildas-lormeau/SingleFile.git'`.
